Create how-to-document-with-docusaurus.md

GuillaumeAV opened the pull request:
Est-ce que c'est complet et exact ? Il doit manquer des choses !

GuillaumeAV committed:
Create how-to-document-with-docusaurus.md

Est-ce que c'est complet et exact ? Il doit manquer des choses !

srosset81 commented:
Super ! Je laisse @ReX-AV-Gab apporter sa contribution et ensuite je ferai le review.

Pensez à mettre un lien dans la sidebar vers ce document ;-)

ReX-AV-Gab commented:
Je crois que je fais un blocage sur la sidebar, je ne la visualise pas pas pas !

srosset81 commented:
Tu ne la visualise pas, c'est-à-dire ?
https://github.com/assemblee-virtuelle/semapps/blob/GuillaumeAV-doc-docusaurus/website/sidebars.js

ReX-AV-Gab commented:
J'avais bien accédé au code mais je n'avais compris que c'était la sidebar de Gouvernance. J'ai peut-être été perturbé par le fait que je n'ai pas vu governance/ hello alors que j'y ai bien vu governance/team !

ReX-AV-Gab commented:
Où met-on la présentation du cadre de notre démarche de documentation?
Dans le readme.md ?
On pourrait y indiquer nos choix en termes de structuration de la documentation, des cibles de la documentation et cetera, et cetera !
Je me redis qu'il serait bien de s'attarder un peu sur l'identité (Caractère de ce qui, sous des dénominations ou des aspects divers, ne fait qu'un ou ne représente qu'une seule et même réalité [CNRTL]) de SemApps !

srosset81 commented:
Qu'en dites-vous @bouviermullerp @GuillaumeAV ? A mon avis il faut bien différencier entre ce qui est documentation interne (donc à mettre plutôt dans Roam Research) et ce qui est pour le site web. 

Concernant le fichier READMD.me à la racine de ce repo, il est destiné à se réduire à sa plus simple expression (juste quelques liens vers le site web ?). Je ne pense pas qu'il faille y ajouter des choses.

GuillaumeAV commented:
Hey ! 
@ReX-AV-Gab l'objet de cette pull request c'est ce commit : 
https://github.com/assemblee-virtuelle/semapps/commit/1f6a33efd1b40d70f577cc7a217e2acd7bde1488 

Tu peux le voir dans "Files changed" également.
Mais à l'heure actuelle, il n'est pas présent dans le docusaurus .. Si j'avais édité le how-to je pense qu'il serait apparu mais il me semblait qu'il fallait un contenu plus générique sur le how-to...

GuillaumeAV commented:
@srosset81 tu suggères je crois d'héberger le tuto concernant la documentation sur le roam research ? Personnellement ca me semblerait pertinent en effet ...

srosset81 commented:
> @srosset81 tu suggères je crois d'héberger le tuto concernant la documentation sur le roam research ? Personnellement ca me semblerait pertinent en effet ...

Non je parlais du "cadre de démarche de documentation" évoqué par Gabriel.
Il me semble que ce que tu as fait à sa place sur le site web.

srosset81 commented:
@ReX-AV-Gab On attend toujours ta bonification de ce que Guillaume a fait ! :-)

ReX-AV-Gab commented:
Je crois que tout ça est au-dessus de mes compétences actuelles
J'ai modifié le fichier  semapps/website/docs/governance/team.md, j'ai cru merger et voir le résultat de ces modifs sur https://website-32fcoc5ln.now.sh/docs/governance/team et là je ne vois plus ces modifs.
Je ne sais plus à quelle branche me vouer

srosset81 commented:
Le fichier est sur la branche `GuillaumeAV-doc-docusaurus`, il y a un lien vers elle tout en haut de cette page.
Tu peux aussi modifier ce qu'a fait Guillaume en allant dans l'onglet "Files changed" de cette PR, et en éditant le fichier de cette manière:
![Capture d’écran de 2020-04-07 08-45-09](https://user-images.githubusercontent.com/17931931/78638250-3f4de900-78ac-11ea-8ffb-48067903eae8.png)

ReX-AV-Gab commented on website/docs/participate/how-to-document-with-docusaurus.md:
```diff
@@ -0,0 +1,44 @@
+
+# Ou trouver la documentation docusaurus
+- https://github.com/assemblee-virtuelle/semapps/tree/master/website/docs
+  - On y trouve les **fichiers de documentation** (gouvernance, participer, documentation) sont dans website/docs 
```
=> j'y ai trouvé les répertoires governance, packages et participate et le fichier style-guide.md

ReX-AV-Gab commented on website/docs/participate/how-to-document-with-docusaurus.md:
```diff
@@ -0,0 +1,44 @@
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/docusaurus.config.js
+  - On y trouve le **menu**, le **header** et le **footer**
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/src/pages/index.js
+  - On y trouve les textes et images de la page d'accueil
```
=> On y trouve le contenu du corps de la page d'accueil et du bouton (Commencer) ?

GuillaumeAV replied:
J'ai intégré ta proposition de modif !

ReX-AV-Gab commented on website/docs/participate/how-to-document-with-docusaurus.md:
```diff
@@ -0,0 +1,44 @@
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/src/pages/index.js
+  - On y trouve les textes et images de la page d'accueil
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/sidebars.js
+  - On y trouve les pages apparaissant dans la sidebar
```
=> j'en vois qu'une seule, je suppose que les autres pages seront mises au même endroit (sidebars.js)

GuillaumeAV replied:
J'ai intégré ta proposition de modif !

ReX-AV-Gab commented on website/docs/participate/how-to-document-with-docusaurus.md:
```diff
@@ -0,0 +1,44 @@
+- https://github.com/assemblee-virtuelle/semapps/tree/master/website/docs
+  - On y trouve les **fichiers de documentation** (gouvernance, participer, documentation) sont dans website/docs 
+- https://github.com/assemblee-virtuelle/semapps/blob/master/website/docusaurus.config.js
+  - On y trouve le **menu**, le **header** et le **footer**
```
=> On y trouve le menu (navbar), le header (title et tagline) et le footer

GuillaumeAV replied:
J'ai intégré ta proposition de modif !

ReX-AV-Gab commented on website/docs/participate/how-to-document-with-docusaurus.md:
```diff
@@ -0,0 +1,44 @@
+
+## Modifier une page du docusaurus**
+Rendez-vous à l'endroit pertinent en vous référant au paragraphe ci-dessus. 
+- Si c'est une modification mineure, une faute d'orthographe par exemple, effectuez la modification en éditant la page, puis "commitez" sur la branche __Master__. Décrivez votre modification à l'endroit réservé à cet effet. 
```
=> Si c'est une modification mineure, une faute d'orthographe par exemple, effectuez la modification en éditant la page depuis la branche __Master__ puis "commitez". Décrivez votre modification à l'endroit réservé à cet effet. 
        Pour l'exemple, j'ai "Commit directly to the master branch" la modif "title: Comment participerons-nous ???" (j'avais déjà fait avec apparition puis disparition de la modif dans le wiki)

GuillaumeAV replied:
Je crains que ca complexifie le message ta formulation ...

ReX-AV-Gab commented on website/docs/participate/how-to-document-with-docusaurus.md:
````diff
@@ -0,0 +1,44 @@
+  title: [le titre de votre page]
+  ---
+  ```
+- Commitez et faites un pull request sur une branche dédiée. 
````
=> J'ai cru qu'il fallait ajouter les " ```" bêtement - Penser à ajouter l'extension md.

srosset81 replied:
Le nom du fichier markdown doit aussi ne pas contenir d'espaces, ni de caractères spéciaux. Privilégier les minuscules.

ReX-AV-Gab committed:
Create Code-social

Pour l'exercice demandé par le formateur
Avec un Commit directly to the GuillaumeAV-doc-docusaurus branch.

ReX-AV-Gab committed:
Create Code-social.md

Avec l'extension cette fois-ci

ReX-AV-Gab committed:
Delete Code-social

srosset81 commented:
@GuillaumeAV tu as mergé ? 
J'attendais les corrections de Gabriel pour faire mon review :-/
Il manque en plus le lien dans la sidebar...
